apm-sniffer/apm-sdk-plugin/spring-plugins/mvc-annotation-5.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/spring/mvc/v5/InvokeInterceptor.java

```diff
@@ -41,6 +52,23 @@ public void beforeMethod(final EnhancedInstance objInst,
                       .put(RESPONSE_KEY_IN_RUNTIME_CONTEXT, new ReactiveResponseHolder(exchange.getResponse()));
         ContextManager.getRuntimeContext()
                       .put(REQUEST_KEY_IN_RUNTIME_CONTEXT, new ReactiveRequestHolder(exchange.getRequest()));
+
+        ContextCarrier carrier = new ContextCarrier();
+        CarrierItem next = carrier.items();
+        HttpHeaders headers = exchange.getRequest().getHeaders();
+        while (next.hasNext()) {
+            next = next.next();
+            List<String> header = headers.get(next.getHeadKey());
+            if (header != null && header.size() > 0) {
+                next.setHeadValue(header.get(0));
+            }
+        }
+
+        AbstractSpan span = ContextManager.createEntrySpan(exchange.getRequest().getURI().getPath(), carrier);
```

Comment:
Why are you creating a new span?

Reply:
beacuse spring webflux chang httpCode after org.apache.skywalking.apm.plugin.spring.mvc.commons.interceptor.AbstractMethodInterceptor, so need create new span to collect httpCode.

Reply:
This is not the right way to change.  If you create 2 entry span(including the existing one), one request would be counted two, and the latency would be wrong.

Reply:
Yes, this is a problem. And I found that tomcat and springmvc plugin have the same problem. How should we fix. createLocalSpan in the springmvc plugin?

Reply:
No, tomcat and springmvc in one thread, the core will ignore the tomcat automatically.

Reply:
If one component goes to async mode, use async API to close span in another thread. Read doc, please.

Reply:
> If one component goes to async mode, use async API to close span in another thread. Read doc, please.

Maybe delivery the threadlocal when the thread switch can solve this problem.

Reply:
Please don't guess. We have supported other reactor mode long time ago, there are explicit document in the plugin dev doc to explain the APIs, and we have plugins for async already.

Reply:
Thanks, this is not the right change.